internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -721,6 +709,39 @@ func (a *Adapter) createMissingReleasesForReleasePlans(application *applicationa
 	return nil
 }
 
+// createAutomatedRelease  creates a new release for a given releasePlan and snapshot and marks it as automated.
+// In case the Releases can't be created or their status can't be updated, an error will be returned.
+func (a *Adapter) createAutomatedRelease(releasePlan *releasev1alpha1.ReleasePlan, snapshot *applicationapiv1alpha1.Snapshot) error {
+	newRelease := release.NewReleaseForReleasePlan(a.context, releasePlan, snapshot)
+	err := retry.OnError(retry.DefaultRetry, func(_ error) bool { return true }, func() error {
+		err := a.client.Create(a.context, newRelease)
+		if err != nil {
+			return err
```

Comment:
Do we want to be more explicit here? Providing message similar to:
Could not create new release, snapshot.Name, releasePlan.name.